src/components/scenes/EdgeLoginScene.tsx

```diff
@@ -1,242 +1,158 @@
-      message = sprintf(s.strings.edge_description_warning, this.props.lobby.loginRequest.displayName)
+    if (lobby && lobby.loginRequest && lobby.loginRequest.appId === '') {
+      message = sprintf(s.strings.edge_description_warning, lobby.loginRequest.displayName)
     }
     return (
-      <View style={styles.body}>
-        <Text style={styles.bodyText}>{message}</Text>
+      <View style={styles.warning}>
+        <Alert title={s.strings.string_warning} message={message} type="warning" numberOfLines={6} />
       </View>
     )
-  }
+  })
 
-  renderButtons() {
-    const { navigation } = this.props
+  const renderButtons = useHandler(() => {
     const handleDecline = () => navigation.goBack()
-
-    if (this.props.isProcessing) {
+    if (isProcessing) {
       return (
         <View style={styles.buttonsProcessing}>
-          <ActivityIndicator color={THEME.COLORS.ACCENT_MINT} />
+          <ActivityIndicator color={theme.primaryText} size="large" />
         </View>
       )
     }
-    if (this.props.error) {
+    if (error) {
       return (
-        <View style={styles.buttonContainer}>
-          <View style={styles.buttons}>
-            <SecondaryButton style={styles.cancelSolo} onPress={handleDecline}>
-              <SecondaryButton.Text>{s.strings.string_cancel_cap}</SecondaryButton.Text>
-            </SecondaryButton>
-          </View>
+        <View style={styles.buttons}>
+          <MainButton label={s.strings.string_cancel_cap} type="escape" onPress={handleDecline} />
         </View>
       )
     }
     return (
-      <View style={styles.buttonContainer}>
-        <View style={styles.buttons}>
-          <SecondaryButton style={styles.cancel} onPress={handleDecline}>
-            <SecondaryButton.Text>{s.strings.string_cancel_cap}</SecondaryButton.Text>
-          </SecondaryButton>
-          <PrimaryButton style={styles.submit} onPress={this.props.accept}>
-            <PrimaryButton.Text>{s.strings.accept_button_text}</PrimaryButton.Text>
-          </PrimaryButton>
-        </View>
+      <View style={styles.buttons}>
+        <MainButton marginRem={[1, 0, 1, 0]} label={s.strings.accept_button_text} onPress={handlePress} />
+        <MainButton label={s.strings.string_cancel_cap} type="escape" onPress={handleDecline} />
       </View>
     )
-  }
+  })
 
-  renderImage() {
-    if (this.props.lobby && this.props.lobby.loginRequest && this.props.lobby.loginRequest.displayImageUrl) {
-      return <FastImage style={styles.image} resizeMode="contain" source={{ uri: this.props.lobby.loginRequest.displayImageUrl }} />
+  const renderImage = useHandler(() => {
+    if (lobby && lobby?.loginRequest && lobby?.loginRequest?.displayImageUrl) {
+      return <Image source={theme.primaryLogo} style={styles.logo} resizeMode="contain" />
     }
     return null
-  }
+  })
 
-  renderHeader() {
+  const renderHeader = useHandler(() => {
```

Comment:
This `useHandler` approach is not necessary, since these are not event handlers, but rather helpers for the render function. You can either drop the `useHandler` wrapper and leave these as plain functions, or you can wrap the blocks with `useMemo` to *actually* get caching logic, so the sections only re-render when real changes happen.